### PR TITLE
Bring back `pwsh.exe` for framework dependent packages to support Start-Job

### LINKED ISSUE #16537: Start-Job fails on PowerShell 7.2.0 when installed as a .NET global tool
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Use this script:
```powershell
Set-StrictMode -Version 3.0
$ErrorActionPreference = "Stop"

$job = Start-Job -ScriptBlock { Write-Host "I am a job." }
Wait-Job -Job $job | Out-Null
Receive-Job -Job $job
```

✔️ When using PowerShell 7.1.5 as a .NET global tool, the script runs successfully. 
✔ When using PowerShell 7.2.0 from the MSI installer, the script runs successfully.
❌ When using PowerShell 7.2.0 as a .NET global tool, the script fails.

The error is:
```
The pwsh executable cannot be found at
     | "C:\Users\<username>\.dotnet\tools\.store\powershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe"
```
It's true that the file does not exist:
```
> get-childitem *pwsh*

    Directory: C:\Users\<username>\.dotnet\tools\.store\powershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           2/11/2021    09:04         100424 pwsh.deps.json
-a---           2/11/2021    09:10         143736 pwsh.dll
-a---           2/11/2021    09:04            583 pwsh.runtimeconfig.json
```

Using PowerShell 7.1.5, that directory also contains `pwsh.exe`. That file is missing in PowerShell 7.2.0.

### Expected behavior

```console
> pwsh .\test.ps1
I am a job
```

### Actual behavior

```console
> pwsh .\test.ps1
Start-Job: D:\test.ps1:4
Line |
   4 |  $job = Start-Job -ScriptBlock { Write-Host "I am a job." }
     |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The pwsh executable cannot be found at
     | "C:\Users\<username>\.dotnet\tools\.store\powershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe". Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
```

### Error details

```console
Type           : System.Management.Automation.CmdletInvocationException
ErrorRecord    :
    Exception             :
        Type        : System.Management.Automation.PSNotSupportedException
        ErrorRecord :
            Exception             :
                Type    : System.Management.Automation.ParentContainsErrorRecordException
                Message : The pwsh executable cannot be found at "C:\Users\<username>\.dotnet\tools\.sto
re\powershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe".
                          Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being
hosted in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
                HResult : -2146233087
            CategoryInfo          : NotImplemented: (:) [], ParentContainsErrorRecordException
            FullyQualifiedErrorId : NotSupported
        TargetSite  :
            Name          : ThrowTerminatingError
            DeclaringType : System.Management.Automation.MshCommandRuntime, System.Management.Automation,
Version=7.2.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35
            MemberType    : Method
            Module        : System.Management.Automation.dll
        Message     : The pwsh executable cannot be found at "C:\Users\<username>\.dotnet\tools\.store\p
owershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe".
                      Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted
in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
        Source      : System.Management.Automation
        HResult     : -2146233067
        StackTrace  :
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
    TargetObject          :
C:\Users\<username>\.dotnet\tools\.store\powershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe
    CategoryInfo          : NotInstalled: (C:\Users\david.bieg….0\any\win\pwsh.exe:String) [Start-Job],
PSNotSupportedException
    FullyQualifiedErrorId : IPCPwshExecutableNotFound,Microsoft.PowerShell.Commands.StartJobCommand
    InvocationInfo        :
        MyCommand        : Start-Job
        ScriptLineNumber : 4
        OffsetInLine     : 8
        HistoryId        : 1
        ScriptName       : D:\test.ps1
        Line             : $job = Start-Job -ScriptBlock { Write-Host "I am a job." }

        PositionMessage  : At D:\test.ps1:4 char:8
                           + $job = Start-Job -ScriptBlock { Write-Host "I am a job." }
                           +        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        PSScriptRoot     : D:\Code\Playground\ps-playground
        PSCommandPath    : D:\test.ps1
        InvocationName   : Start-Job
        CommandOrigin    : Internal
    ScriptStackTrace      : at <ScriptBlock>, D:\test.ps1: line 4
                            at <ScriptBlock>, <No file>: line 1
TargetSite     :
    Name          : Invoke
    DeclaringType : System.Management.Automation.Runspaces.PipelineBase, System.Management.Automation,
Version=7.2.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35
    MemberType    : Method
    Module        : System.Management.Automation.dll
Message        : The pwsh executable cannot be found at "C:\Users\<username>\.dotnet\tools\.store\powers
hell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe".
                 Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in
other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
Data           : System.Collections.ListDictionaryInternal
InnerException :
    Type        : System.Management.Automation.PSNotSupportedException
    ErrorRecord :
        Exception             :
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : The pwsh executable cannot be found at "C:\Users\<username>\.dotnet\tools\.store\p
owershell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe".
                      Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted
in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
            HResult : -2146233087
        CategoryInfo          : NotImplemented: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : NotSupported
    TargetSite  :
        Name          : ThrowTerminatingError
        DeclaringType : System.Management.Automation.MshCommandRuntime, System.Management.Automation,
Version=7.2.0.500, Culture=neutral, PublicKeyToken=31bf3856ad364e35
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message     : The pwsh executable cannot be found at "C:\Users\<username>\.dotnet\tools\.store\power
shell\7.2.0\powershell\7.2.0\tools\net6.0\any\win\pwsh.exe".
                  Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in
other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
    Source      : System.Management.Automation
    HResult     : -2146233067
    StackTrace  :
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
Source         : System.Management.Automation
HResult        : -2146233087
StackTrace     :
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at System.Management.Automation.Runspaces.Pipeline.Invoke()
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown,
ExecutionOptions options)
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.0
PSEdition                      Core
GitCommitId                    7.2.0
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Fix #16537
For a framework dependent package we removed pwsh.exe recently, but we need it for supporting `Start-Job`. So bring it back.

## PR Context

Customers of .NET 6 SDK docker images reported the issue here: https://github.com/dotnet/dotnet-docker/issues/3302

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
Will fix #16532?

**Review by PaulHigin — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
Marking it as WIP as I am adding some tests.

**Review by daxian-dbw — APPROVED** [review]

File: tools/releaseBuild/azureDevOps/releasePipeline.yml [hunk 1]
```diff
@@ -78,6 +78,31 @@ stages:
...
+  dependsOn: []
+  jobs:
+  - template: templates/release-FxdPackage.yml
+    parameters:
+      jobName: WinFxdPackage
+      displayName: Fxd Package Test Win
+      imageName: windows-latest
+      packageNamePattern: '**/*win-fxdependent.zip'
+
+  - template: templates/release-FxdPackage.yml
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/releasePipeline.yml:92]: Can we include something about validation in the name of the file?

File: build.psm1 [hunk 2]
```diff
@@ -439,10 +439,9 @@ Fix steps:
     # Add --self-contained due to "warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used."
     if ($Options.Runtime -like 'fxdependent*') {
         $Arguments += "--no-self-contained"
-        # The UseAppHost = false property avoids creating ".exe" for the fxdependent packages.
-        # The ".exe" is not a cross-platform executable, but specific to the platform that it was built on.
-        # We do not need to ship that.
-        $Arguments += "/property:UseAppHost=false"
+        # The UseAppHost = true property creates ".exe" for the fxdependent packages.
+        # We need this in the package as Start-Job needs it.
+        $Arguments += "/property:UseAppHost=true"
```
Comment (TravisEz13) [inline build.psm1:444]: I think we removed this because it was not signed.. we should sign it.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
the signing issue is the main thing that needs to be addressed

File: tools/releaseBuild/azureDevOps/releasePipeline.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/releasePipeline.yml:92]: ok I will update the file name

File: build.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply build.psm1:444]: Good point.. let me check the test build.

File: tools/releaseBuild/azureDevOps/releasePipeline.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/releasePipeline.yml]: Updated

File: build.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply build.psm1:444]: Updated the packaging code.

**Comment by adityapatwardhan** [comment]
@TravisEz13 I have kicked off a test build with signing to validate the change. I will remove WIP when I have validated it.

**Comment by adityapatwardhan** [comment]
@TravisEz13 validated that the fxdependent package has pwsh.exe which is signed.

**Review by TravisEz13 — APPROVED** [review]

**Comment by cboulva99** [comment]
Hello, Thank you for the fix @adityapatwardhan , 

So, any idea when will this fix make it in the official  dotnet6 sdk docker images?

Thank you in advance,

BR

**Comment by adityapatwardhan** [comment]
We are trying to get a fix in for the next release of the images.

**Pull request merged**